lms/djangoapps/commerce/api/v1/serializers.py

```diff
@@ -25,9 +31,30 @@ class Meta(object):  # pylint: disable=missing-docstring
         fields = ('name', 'currency', 'price', 'sku', 'expires')
 
 
+def validate_course_id(course_id):
+    """
+    Check that course id is valid and exists in modulestore.
+    """
+    try:
+        course_key = CourseKey.from_string(unicode(course_id))
+    except InvalidKeyError:
+        raise serializers.ValidationError(
+            _("{course_id} is not a valid course key.").format(
+                course_id=course_id
+            )
+        )
+
+    if not modulestore().has_course(course_key):
```

Comment:
This should be checking CourseOverviews, not the modulestore directly. I've filed [ECOM-2276](https://openedx.atlassian.net/browse/ECOM-2276) to fix this.